source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOVisibilityManager.mm

```diff
@@ -0,0 +1,118 @@
+//
+//  ACOVisibilityManager.mm
+//  AdaptiveCards
+//
+//  Copyright © 2021 Microsoft. All rights reserved.
+//
+//
+
+#import "ACOVisibilityManager.h"
+#import "ACREnums.h"
+#import "ACRSeparator.h"
+
+@implementation ACOVisibilityManager
+
+- (void)hideView:(UIView *)viewToBeHidden arrangedSubviews:(NSArray<UIView *> *)subviews
+{
+    if (!subviews || !viewToBeHidden) {
+        return;
+    }
+
+    if (!viewToBeHidden.isHidden) {
+        viewToBeHidden.hidden = YES;
+    }
+
+    // get the index of the target view
+    // if the target view is not the tail,
+    // check to the right, and if the right separator and not hidden,
+    // hide the separator
+    // if the target view is tail, check to left, and if the left is separator and not hidden,
+    // nhide the separator
+    NSInteger index = [subviews indexOfObject:viewToBeHidden];
+    if (index + 1 < subviews.count) {
+        NSInteger nextViewIndex = index + 1;
+        UIView *nextView = subviews[nextViewIndex];
+        if ([nextView isKindOfClass:[ACRSeparator class]]) {
+            if (!nextView.isHidden) {
+                nextView.hidden = YES;
+            }
+        }
+    } else if (index - 1 >= 0) {
+        NSInteger prevViewIndex = index - 1;
+        UIView *prevView = subviews[prevViewIndex];
+        if ([prevView isKindOfClass:[ACRSeparator class]]) {
+            if (!prevView.isHidden) {
+                prevView.hidden = YES;
+            }
+        }
+    }
+
+    // only cares for tailing padding at the moment
+    // if column width is "auto", and all of its children hidden,
+    // hide column, if is "stretch", then don't hide it.
```

Comment:
why is a hidden stretch column expected to occupy space? are we trying to mimic the behaviour of visibility invisible and collapsed with this behavior?

Reply:
David said that auto takes the consideration of content size. So if its children are all hidden, the column's content will become zero and doesn't occupy the space. With the stretch, the content size is not considered, and it occupies space regardless. And I think he is correct, this is in line with what we have defined for column width.

Reply:
if this was defined and accepted by the team then it's ok